src/libraries/System.Net.Sockets/tests/FunctionalTests/SendFile.cs

```diff
@@ -156,6 +164,39 @@ public async Task SendFile_NoFile_Succeeds(bool useAsync, bool usePreBuffer, boo
             Assert.Equal(0, client.Available);
         }
 
+        [Theory]
+        [InlineData(false, false)]
+        [InlineData(false, true)]
+        [InlineData(true, false)]
+        [InlineData(true, true)]
+        public async Task SendFileAsync_NoFile_Succeeds(bool usePreBuffer, bool usePostBuffer)
```

Comment:
A lot of other socket tests (e.g. SendReceive, connect, etc) are generalized for different "modes" (sync, various async, span vs no span, etc) by deriving from SocketTestHelperBase and then instantiating appropriate versions of the class. We don't do this currently with SendFile, but since we are adding all these overloads, it probably makes sense to. That means adding SendFile to SocketHelperBase and providing relevant implementations on the derived classes, e.g. SocketHelperApm etc.

Reply:
I think rebasing tests on `SocketTestHelperBase` is a must-have with this change.

Looks like most of the tests can be refactored to the base class (`SendFile<T> : SocketTestHelperBase<T> where T : SocketHelperBase`), the ones that are specific to an (APM/Sync/Task) implementation can be moved to the subclass.

Reply:
Makes sense, but can we do this in a different PR? There is quite some rebasing needed as
https://github.com/dotnet/runtime/blob/f066972c7aa9f9147347623617e789e70aff3d3f/src/libraries/System.Net.Sockets/tests/FunctionalTests/SendFile.cs#L15 already inherits from `FileCleanupTestBase` (I guess this may be a reason why it wasn't done so far.)